src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/Auxiliary2AzureSearchCommand.cs

```diff
@@ -271,7 +271,7 @@ private void CleanDownloadData(DownloadData data)
 
             foreach (var id in data.Keys.ToList())
             {
-                var isValidId = PackageIdValidator.IsValidPackageId(id);
+                var isValidId = id.Length <= PackageIdValidator.MaxPackageIdLength && PackageIdValidator.IsValidPackageId(id);
```

Comment:
Should we update `IsValidPackageId` directly?

Reply:
Client code. Not worth since we have unit test coverage.